CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLHarvester.cc

```diff
@@ -0,0 +1,158 @@
+/****************************************************************************
+ *
+ * This is a part of PPS offline software.
+ * Authors:
+ *   Edoardo Bossini
+ *   Piotr Maciej Cwiklicki
+ *   Laurent Forthomme
+ *
+ ****************************************************************************/
+
+#include "DQMServices/Core/interface/DQMEDHarvester.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+#include "CalibPPS/TimingCalibration/interface/TimingCalibrationStruct.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "CondFormats/PPSObjects/interface/PPSTimingCalibration.h"
+
+//------------------------------------------------------------------------------
+
+class PPSTimingCalibrationPCLHarvester : public DQMEDHarvester {
+public:
+  PPSTimingCalibrationPCLHarvester(const edm::ParameterSet&);
+  void beginRun(const edm::Run&, const edm::EventSetup&) override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  void dqmEndJob(DQMStore::IBooker&, DQMStore::IGetter&) override;
+  edm::Service<cond::service::PoolDBOutputService> poolDbService_;
```

Comment:
The service handle should be instantiated separately in the member functions of an EDModule instead of storing as a member.

Reply:
@makortel Many thanks for your review!
So would you suggest having a first instance built in the scope of the constructor (only to check if the service is loaded), and another one used in the dqmEndJob?